libs/features/mas/src/controllers/height-sync-controller.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Reactive controller for coordinating height synchronization
+ * across plans-v2 cards in a collection.
+ *
+ * Follows Lit reactive controller pattern:
+ * https://lit.dev/docs/composition/controllers/
+ */
+export class HeightSyncController {
+    host;
+    syncScheduled = false;
+    resizeObserver;
+    mutationObserver;
+
+    constructor(host) {
+        this.host = host;
+        this.host.addController(this);
+    }
+
+    async hostConnected() {
+        // Wait for host to be ready
+        await this.host.updateComplete;
+
+        // Schedule initial sync
+        this.scheduleSync();
+
+        // Watch for card additions/removals
+        this.mutationObserver = new MutationObserver(() => this.scheduleSync());
+        this.mutationObserver.observe(this.host, {
+            childList: true,
+            subtree: true,
+        });
+
+        // Watch for viewport resize
+        if (window.ResizeObserver) {
+            this.resizeObserver = new ResizeObserver(() => this.scheduleSync());
+            this.resizeObserver.observe(this.host);
+        }
+
+        // Listen for tab visibility changes
+        if (this.host.closest('[role="tabpanel"]')) {
+            document.addEventListener('visibilitychange', this.handleVisibilityChange);
+        }
+
+        // Listen for card ready events
+        this.host.addEventListener('merch-card-ready', this.handleCardReady);
+    }
+
+    hostDisconnected() {
+        this.mutationObserver?.disconnect();
+        this.resizeObserver?.disconnect();
+        document.removeEventListener('visibilitychange', this.handleVisibilityChange);
+        this.host.removeEventListener('merch-card-ready', this.handleCardReady);
+    }
+
+    handleVisibilityChange = () => {
+        if (document.visibilityState === 'visible') {
+            this.scheduleSync();
+        }
+    };
+
+    handleCardReady = (event) => {
+        if (event.detail?.variant === 'plans-v2') {
+            this.scheduleSync();
+        }
+    };
+
+    scheduleSync() {
+        // Only sync on desktop
+        if (!window.matchMedia('(min-width: 768px)').matches) {
+            return;
+        }
+
+        // Debounce multiple calls
+        if (this.syncScheduled) return;
+
+        this.syncScheduled = true;
+        requestAnimationFrame(async () => {
```

Comment:
btw, requestAnimationFrame doesn't expect an async function, it will not be awaited.